sympy/functions/elementary/integers.py

```diff
@@ -243,15 +286,46 @@ def _eval_Eq(self, other):
     def __lt__(self, other):
         if self.args[0] == other and other.is_real:
             return S.false
+        if other is S.Infinity and self.is_finite:
+            return S.true
+
+        if S(other).is_integer:
+            return self.args[0] <= other - 1
+
         return Lt(self, other, evaluate=False)
 
+    def __gt__(self, other):
+        if self.args[0] == other and other.is_real:
+            return S.false
+        if other is S.NegativeInfinity and self.is_finite:
+            return S.true
+
+        if S(other).is_integer:
+            return self.args[0] > other
+
+        return Gt(self, other, evaluate=False)
+
     def __ge__(self, other):
         if self.args[0] == other and other.is_real:
             return S.true
-        if other is S.NegativeInfinity and self.is_real:
+        if other is S.NegativeInfinity and self.is_finite:
             return S.true
+
+        if S(other).is_integer:
+            return self.args[0] > other - 1
+
         return Ge(self, other, evaluate=False)
 
+    def __le__(self, other):
+        if self.args[0] == other and other.is_real:
+            return S.false
+        if other is S.Infinity and self.is_finite:
+            return S.true
+
+        if S(other).is_integer:
```

Comment:
I think it makes sense to `sympify` `other` at the beginning of the method (in all comparison methods).

I get the interesting behavior:
```
In [56]: floor(7, evaluate=False) >= 7
Out[56]: ⌊7⌋ ≥ 7

In [57]: floor(7, evaluate=False) >=6
Out[57]: True

In [58]: floor(7, evaluate=False) >=8
Out[58]: False
```
on current master and I think it relates to this. (`self.args[0] != other` since it is not sympified.)